README.md

```diff
@@ -72,6 +73,184 @@ $myEntity = Stub::extends($myEntity, ['property1' => 'value']);
 self::assertSame('Hello world!', $myEntity->salute());
 ```
 
+### Test Checks
+
+To run test checks over your test suite, you can for example create umbrella test that allows you to configure and run desired checks.
+Data provider yields checks encapsulated in factory functions in order to allow to pass current test instance and isolate the run. 
+
+```php
+<?php
+
+use Cdn77\TestUtils\TestCheck\TestCheck;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @group       integration
+ * @testedClass none
+ */
+final class SuiteComplianceTest extends TestCaseBase
+{
+    /** 
+     * @dataProvider providerChecks
+     *
+     * @param callable(self): TestCheck $checkFactory 
+     */
+    public function testChecks(callable $checkFactory) : void
+    {
+        $check = $checkFactory($this);
+        $check->run();
+    }
+
+    /** @return Generator<string, array{callable(self): TestCheck}> */
+    public function providerChecks() : Generator
+    {
+        $testDir = ROOT_PROJECT_DIR . '/tests';
+        $testFilePathNames = \Symfony\Component\Finder\Finder::create()
+            ->in($testDir)
+            ->files()
+            ->name('*Test.php');
+
+        yield 'Test check title' => [
+            static fn (self $suite) => new TestCheckClass(suite, $testFilePathNames [, ...])),
+        ];
+        
+        ...
+    }
+}
+```
+
+### Every test has group
+
+Asserts that all tests have a `@group` annotation 
+
+:x:
+```php
+final class FooTest extends TestCase
+```
+
+:white_check_mark:
```

Comment:
můžeme dát green místo white pls? :D

Reply:
![image](https://user-images.githubusercontent.com/327717/98401882-add36f00-2066-11eb-9f30-3b7777e57f13.png)

Reply:
lol asi je to system specific :D take heavy? :D protože u mě ta white na bílém pozadí basically není vidět :D 

![image](https://user-images.githubusercontent.com/972493/98441789-db6cf680-2100-11eb-898e-a22ad0e6b0e4.png)

Reply:
ummm

chtel jsem neco vyraznyho, u tebe to neni ani jedno :D 

https://gist.github.com/rxaviers/7360908 mrkni sem, any tips?

Reply:
nejlepš je ta co jsem měl ve screenu heavy_check_mark ... i podle názvu to je ok :D

Reply:
fakin unified rendering